internal/rbd/controllerserver.go

```diff
@@ -424,6 +430,19 @@ func (cs *ControllerServer) CreateVolume(
 		return nil, err
 	}
 
+	// Apply Qos parameters to rbd image.
+	err = rbdVol.ApplyQOS(ctx)
+	if err != nil {
+		log.DebugLog(ctx, "failed apply QOS for rbd image: %v", err)
+
+		return nil, status.Error(codes.Internal, err.Error())
```

Comment:
note that a CSI function like `CreateVolume` needs to return a `status.Error()` or `status.Errorf()` generated type of error. Functions like `createBackingImage()` should return a 'normal' error. This is not in-sync with the error returned from `SaveQOS()` in the call below.

Reply:
Get